torch/testing/_internal/common_methods_invocations.py

```diff
@@ -303,6 +309,52 @@ def sample_inputs(self, device, dtype, requires_grad=False, **kwargs):
             samples = self.sample_inputs_func(self, device, dtype, requires_grad)
         return samples
 
+    def sample_inputs(self, device, dtype, requires_grad=False, **kwargs):
+        """Generates a sequence of SampleInputs.
+
+        The OpInfo keeps track the the sample which is currently yielded: wrap
+        your code inside a with op.decorate_errors() to get better exception
+        messages which include the current sample.
+
+        If subclasses want to customize the generation of inputs, they should
+        override generate_sample_inputs(). On the contrary, end users should
+        always call sample_inputs().
+        """
+        if self._current_sample is not None:
+            raise ValueError('Cannot call op.sample_inputs() while another '
+                             'iteration is in progress')
+        samples = self.generate_sample_inputs(device, dtype, requires_grad, **kwargs)
+        for i, sample in enumerate(samples):
+            assert isinstance(sample, SampleInput)
+            # make sure that every SampleInput has a sensible name: if the
+            # name was not given, use the index instead
+            if sample.name is None:
```

Comment:
Using @kshitij12345's work I'd ditch the enumerate here and just print the sample input, named or unnamed